docs/proposals/job.md

```diff
@@ -0,0 +1,135 @@
+# Job Controller
+
+## Abstract
+A proposal for implementing a new controller - Job controller, which will be responsible
+for managing pod(s) that require to run on demand.
+
+Several existing issues and PRs were already created regarding that particular subject:
+* Job Controller [#1624](https://github.com/GoogleCloudPlatform/kubernetes/issues/1624)
+* New Job resource [#7380](https://github.com/GoogleCloudPlatform/kubernetes/pull/7380)
+
+
+## Use Cases
+1. Be able to start of one or several pods tracked as a single entity.
+1. Be able to implement heavy-computation model, eg. MapReduce.
+1. Be able to get the actual job status.
+1. Be able to limit the execution time for a job.
+
+
+## Motivation
+Jobs are needed for executing multi-pod computation "to completion", a good example
+here would be the ability to implement a MapReduce or Hadoop style workload.
+Additionally this new controller should take over pod management logic we currently
+have in certain OpenShift controllers, namely build controller.
+
+
+## Implementation
+Job controller, is similar to replication controller in that they manage pods
+representing a `Job`.  This implies they will follow the same controller framework
+that replication controllers already defined.  The biggest difference between
+`Job` and `ReplicationController` objects is the `RestartPolicy` which is required
+to always take value of `RestartPolicyNever` or `RestartOnFailure`.  The main
+responsibility of a job controller will be keeping that the number of pods will
```

Comment:
I'll move the sentence about the purpose of the Job controller before the `RestartPolicy`, as that is stronger differentiator.